taf/updater/updater_pipeline.py

```diff
@@ -399,14 +401,25 @@ def check_if_local_repositories_clean(self):
                     return UpdateStatus.SUCCESS
                 else:
                     if auth_repo.something_to_commit():
-                        raise RepositoryNotCleanError(auth_repo.name)
+                        if self.force:
+                            taf_logger.info(
+                                f"Cleaning and resetting repository {auth_repo.name} for forced update"
+                            )
+                            auth_repo.clean_and_reset()
+                        else:
+                            raise RepositoryNotCleanError(auth_repo.name)
```

Comment:
Let's extend these exceptions, so that if they are by the updater, we print a message saying that the updater can be run with the `--force` flag.